File: lab1/fp_SimUDuck/MallardDuck.cpp
#include "stdafx.h"
#include "MallardDuck.h"

MallardDuck::MallardDuck()
	: Duck(FlyBehaviors::GetFlyWithWings(),
		QuackBehaviors::GetQuackBehavior(),
		DanceBehaviors::GetWaltzDanceBehavior())
{
}

void MallardDuck::Display() const
{
	std::cout << "I'm mallard duck" << std::endl;
}

File: lw1_fn/src/duck/MallardDuck.php
<?php

class MallardDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(
            FlyBehavior\getFlyWithWings(),
            QuackBehavior\getQuackBehavior(),
            DanceBehavior\getWaltzDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m mallard duck!' . PHP_EOL;
    }
}